2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md

```diff
@@ -0,0 +1,128 @@
+---
+
+layout: col-sidebar
+title: "M6: Insufficient Input/Output Validation"
+---
+
+# Threat Agents
+
+**Application Specific**
+
+If the application fails to properly validate and sanitize data received from external sources like user inputs or network data. This oversight can lead to various security vulnerabilities, such as SQL injection, Command Injection or cross-site scripting (XSS) attacks. 
+
+These attacks can lead to unauthorized data access, manipulation, and potentially compromise the entire system. 
+
+Inadequate output validation can result in data corruption or presentation vulnerabilities, allowing malicious actors to inject malicious code or manipulate sensitive information displayed to users.
+
+# Attack Vectors	
+
+**Exploitability HIGH**
+
+Insufficient input/output validation exposes our application to critical attack vectors, including SQL injection, XSS, command injection, path traversal, HTML injection, and remote file inclusion. These vulnerabilities can lead to unauthorized access, data manipulation, code execution, and compromise of the entire system. 
```

Comment:
I would remove remote file inclusion which is very rare on the client side due to the languages commonly in use in mobile apps.

Reply:
ok, will do